Sources/GRPC/Interceptor/ServerInterceptorContext.swift

```diff
@@ -56,6 +56,21 @@ public struct ServerInterceptorContext<Request, Response> {
     return self.pipeline.path
   }
 
+  /// A 'UserInfo' dictionary.
+  ///
+  /// - Important: While `UserInfo` has value-semantics, this property retrieves from, and sets a
+  ///   reference wrapped `UserInfo`. The contexts passed to the service provider share the same
+  ///   reference. As such this may be used as a mechanism to pass information between interceptors
+  ///   and service providers.
+  public var userInfo: UserInfo {
+    get {
+      return self.pipeline.userInfoRef.value
+    }
+    nonmutating set {
+      self.pipeline.userInfoRef.value = newValue
+    }
+  }
```

Comment:
I don't love this... a few alternatives I considered:
- Give `UserInfo` reference semantics... kinda gross and unexpected since it's dictionary-like
- Expose a `UserInfoRef`, a little clearer in its intention, users would have to access the underlying value directly: `userInfoRef.value.foo = "..."` or define their extensions on `UserInfoRef`.

Reply:
I also think this is fine, but I think we need to document the threading requirements here.